lib/src/realm_class.dart

```diff
@@ -656,3 +702,17 @@ class MigrationRealm extends DynamicRealm {
 
   MigrationRealm._(Realm realm) : super._(realm);
 }
+
+/// The signature of a callback that will be executed while the Realm is opened asynchronously with [Realm.open].
+/// This is the registered callback onProgressCallback to receive progress notifications while the download is in progress.
+///
+/// * syncProgress - an object of [SyncProgress] that contains `transferredBytes` and `transferableBytes`.
+/// {@category Realm}
+typedef ProgressCallback = void Function(SyncProgress syncProgress);
+
+/// @nodoc
+extension $CancellableCompleterExtension<T> on Completer<T> {
```

Comment:
This is not currently used, but I think it is nice to have it in order to convert `Completer` to `CancellableCompleter`.

Reply:
Removed